src/sentry/api/validators/sentry_apps/schema.py

```diff
@@ -0,0 +1,261 @@
+from __future__ import absolute_import
+
+from jsonschema import validate as json_schema_validate
+
+SCHEMA = {
+    'type': 'object',
+
+    'definitions': {
+
+        # Property Types
+
+        'uri': {
+            'type': 'string',
+            'format': 'uri',
+        },
+
+        'options': {
+            'type': 'array',
+            'items': {
+                'type': 'array',
+                'minItems': 2,
+                'maxItems': 2,
+                'items': [
+                    {'type': 'string'},
+                    {'anyOf': [
+                        {'type': 'string'},
+                        {'type': 'number'},
+                    ]}
+                ]
+            }
+        },
+
+        'fieldset': {
+            'type': 'array',
+            'minItems': 1,
+            'items': [
+                {
+                    'anyOf': [
+                        {'$ref': '#/definitions/select'},
+                        {'$ref': '#/definitions/text'},
+                    ],
+                },
+            ],
+        },
+
+        # Composable Components
+
+        'select': {
+            'type': 'object',
+            'properties': {
+                'type': {
+                    'type': 'string',
+                    'enum': ['select'],
+                },
+                'label': {
+                    'type': 'string',
+                },
+                'name': {
+                    'type': 'string',
+                },
+                'uri': {
+                    '$ref': '#/definitions/uri',
+                },
+                'options': {
+                    '$ref': '#/definitions/options',
+                },
+            },
+            'required': ['type', 'name', 'label'],
+            'oneOf': [
+                {'required': ['uri']},
+                {'required': ['options']},
+            ],
+        },
+
+        'text': {
+            'type': 'object',
+            'properties': {
+                'type': {
+                    'type': 'string',
+                    'enum': ['text'],
+                },
+                'label': {
+                    'type': 'string',
+                },
+                'name': {
+                    'type': 'string',
+                },
```

Comment:
Can fields be marked as required through an attribute?

Reply:
Doesn't seem like it 😒 

https://json-schema.org/understanding-json-schema/reference/object.html#required